Misc/NEWS.d/next/Library/2025-03-01-15-00-00.gh-issue-110067.1ad3as.rst

```diff
@@ -0,0 +1 @@
+Make max heap functions public.
```

Comment:
This requires a true what's new entry + indicate which functions are available. I would also always use `max-heap` with a dash. In addition, we can say that the private names are kept for backwards compatibility (something like "Previous underscored naming is kept for backwards compatibility")